### ISSUE TITLE
Reload upon activity resume

### ISSUE THREAD

**Issue by KopiasCsaba** [FossifyOrg/Gallery#284]
### Checklist

- [X] I made sure that there are **no existing issues** - [open](https://github.com/FossifyOrg/Gallery/issues) or [closed](https://github.com/FossifyOrg/Gallery/issues?q=is%3Aissue+is%3Aclosed) - to which I could contribute my information.
- [X] I made sure that there are **no existing discussions** - [open](https://github.com/FossifyOrg/Gallery/discussions) or [closed](https://github.com/FossifyOrg/Gallery/discussions?discussions_q=is%3Aclosed) - to which I could contribute my information.
- [X] I have read the FAQs inside the app (Menu -> About -> FAQs) and my problem isn't listed.
- [X] **I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.**
- [X] This issue contains only one feature request.
- [X] I have read and understood the [contribution guidelines](https://github.com/FossifyOrg/Gallery/blob/master/CONTRIBUTING.md).
- [ ] I optionally [donated](https://fossify.org/donate) to support the Fossify mission.

### Feature description

Upon starting the app or reactivating it (if it was running just switched away and came back), it'd be nice to have an automatic reload of images in that directory. Due to performance reasons I understand if it gets a setting buried somewhere and default off as well.

### Why do you want this feature?

So on graphene OS on pixel phones, thankfully we can use the awesome(ly working) factory camera app the google camera app without the google ecosystem. But, it is hardcoded to open google's photo gallery, so if it's not installed it will not open the gallery from the app just whine. 

What we (including my wife) do now is manually switching to Fossify gallery. But there is an extra step to update the pictures (pull down), and that could be eliminated with this.

I also wonder to rename the package somehow to get it working straight from the camera app, but that's off topic.

### Additional information

I'm very thankful for this project, thank you guys.

**Comment by Aga-C** [comment]
It should already work like this, see the video below (on my two non-rooted devices it works the same):

https://github.com/user-attachments/assets/bcbc142f-7b94-4d45-be29-01ee6820cad4

Are you doing it the same way? If so, maybe it's a problem with loading working too slow. Have you tried to wait a bit more? We have known performance issues related to the image list.

**Comment by KopiasCsaba** [comment]
Dear @Aga-C ,

Thank you for your swift and kind reply, i looked around a bit more to see whats going on, and it is a funny racecondition between the camera app and fossify gallery.

I have made a screenrecording but im on the go, anyhow i switched between the two apps slowly to be visible on the recording. And voila, indeed on activity resume it reloads.

But!

If i make a photo and immediately switch to the galledr app, it refreshes BEFORE the picture is saved back in the camera app. And then it will never show up.

So, steps to reproduce:

- Open f.g., select camera dir
- Open the camera
- Snap a photo and switch back to f.g. as fast as you can.

 (With gesture nav its very easy to switch fast, probably you can't reproduce it with the three button nav)


If you can't reproduce or curious, at night i'll upload a video where you can see it happening.

So i wonder what would be a nice solution. The absolute nerd option would be two settings parameter:
- Extra reload after resume
- Extra reload delay
But it might be confusing for many:)

So maybe an "extra reloads after resume" option that would reload 4 time after resume with one second delay could be something inbetween?

So it would reload asap as the activity is reumed, then 4 more times after a second delay each might be good for most use cases. 

Let me know what do you think!

**Comment by KopiasCsaba** [comment]
Here is a video to show it: https://www.youtube.com/watch?v=w_KBMJ048GI

I also wonder that a loading indicator would be nice to be seen when reloading is in progress when it's just because of the activity resume.

**Comment by Aga-C** [comment]
I could only reproduce it by switching quickly between the apps, at the pace you've shown in the video it was working fine. What device model do you have?

**Comment by naveensingh** [comment]
New items should show up regardless because we have a content observer listening for changes:

https://github.com/FossifyOrg/Gallery/blob/f649b1ef615ab0d41433fdf04ccdda3291a6fecc/app/src/main/kotlin/org/fossify/gallery/activities/SimpleActivity.kt#L21-L32

Evidently, it doesn't seem to be working correctly.

**Comment by KopiasCsaba** [comment]
> I could only reproduce it by switching quickly between the apps, at the pace you've shown in the video it was working fine. What device model do you have?

Yeah, it probably depends on the device and the camera app. I'm using Pixel 7A with the Google Camera. It appears to think around quite a bit before saving the image to the storage apparently.

But it's good news that there is a pub/sub going on although it might be broken at the moment, but then no need to do extra refreshes.

**Comment by paskalito** [comment]
Info:
Oneplus 6
Linage OS 
Googlecamera (same picture vieweing situation)

Happy to have found this:
Not sure if i should still make a new FR, pls tell me if i should:
But the thing i was going to report is.
It never "reloads"/"Advertises" New pictures if you are watching the till then newest one in full screen.
Meaning i cannot see newest pictures until i go to Picture overview and back again.

Made a little video to show the behaviour:

https://github.com/user-attachments/assets/7ad4a848-a617-4f0b-b327-956e0076e429

**Comment by naveensingh** [comment]
> Not sure if i should still make a new FR, pls tell me if i should:

It is related, no need for a new issue :)

**Comment by KopiasCsaba** [comment]
Just now i:
 * I had Camera listing open
 * switched to camera shot 3 images
 * switched back, after some delay they showed up
 * switched back to cam, shot 3 more
 * switched back to listing
 
And those new images never showed up!

So sometimes it works, sometimes it doesnt.

**Comment by KopiasCsaba** [comment]
I was working on something just now, that had to do with the same problem.

So TLDR it seems that there are multiple notifications, the first would show IS_PENDING=1 (without readable image) and then an IS_PENDING=0 would arrive. Unsure of fossify gallery's logic, but maybe thats what is throwing it off.

Here is some proper AI slop that I think might be useful for the dev wanting to fix this.

[details.md](https://github.com/user-attachments/files/28771926/details.md)
